### PR TITLE
Fixes #1351 - State-dependent method bodies part 2

### LINKED ISSUE #1351: State-dependent methods / internal events.
Sometimes a user wants to create behaviour that varies based on state. But if the method has side-effects one needs to set aside a variable to store the result as events in umple currently don't have return types.

```
class Portal {
  statusWord = "";
  description = {"Current status: "+getStatusWord();}
  before getStatusWord {describe();}

  sm {
    closed {
      describe / {setStatusWord("closed");} -> closed;
      open -> open;
    }
    open {
      describe / {setStatusWord("open");} -> open;
      shut -> closed;
    }
  }
}

```

The above is wordy and the user must explicitly code the self transition that might clutter up the diagram, might be error prone, and is awkward. It would be better to be able to define a method in a state thus:

```
class Portal {
  description = {"Current status: "+getStatusWord();}

  sm {
    closed {
      String getStatusWord() {return ("closed");}
      open -> open;
    }
    open {
      String getStatusWord() {return ("open");}
      shut -> closed;
    }
  }
}
```

Any Umple method syntax should work.

The generated code would generate a single method getStatusWord with a case statement that would execute the right block of code depending on the state (with priority given to substate versions overriding higher level state versions. If no method version is defined in a state then a default of doing nothing and returning an empty string, null, zero, false or nothing, depending on the method;s return type.

The above would serve to implement UML's internal event capability.

In a queued or pooled state machine, should such a method be in the queue or be a direct call? I tend to think the former.

The above could be done by 
a) Allowing any event method to return a value initiaally (with consistency enforcement, and default enforcement)
b) Adding the ability to tag any self transition as internal e.g. ~>
c) Allowing an ordinary method signature to implicitly define an internal transition.

### PR DESCRIPTION
**Pull request by pwang347**
## Context
Completes remaining tasks for #1351.
* Different modifiers now generates warning 82 (added associated manual files)
* Top-level state machines now enabled
* Fixed bug with extra space (space is now trimmed in modifiers)

## Tests performed
* Tested more complicated state nesting
* Tested multiple top-level state machines
* Tested the new warning generation

### CODE REVIEW

File: cruise.umple/test/cruise/umple/statemachine/implementation/java/stateDependentMethodDeclaration_2.java.txt [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  {
+    return sm;
+  }
+
+  public void pstatus()
+  {
+    switch(sm)
+    {
+      case init:
+        System.out.println("init");
```
Comment (pwang347) [inline cruise.umple/test/cruise/umple/statemachine/implementation/java/stateDependentMethodDeclaration_2.java.txt:47]: Would probably be a good idea to test injections for methods with return values as well

**Comment by TimLethbridge** [comment]
@pwang347 Are you able to look into the conflicting file UmpleParserTest.java that was also, I think modified recently by another PR. You simply need to merge your changes into what is current. I am not sure why it can't happen automatically.

File: cruise.umple/test/cruise/umple/statemachine/implementation/java/stateDependentMethodDeclaration_2.java.txt (hunk 1 above)
Comment (TimLethbridge) [reply cruise.umple/test/cruise/umple/statemachine/implementation/java/stateDependentMethodDeclaration_2.java.txt:47]: Yes,I agee

File: build/reference/9082ConflictingMethodModifers.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+W082 Conflicting Method Modifiers
+Errors and Warnings 51-99
+noreferences
+
+@@description
+
+<h2>Umple sematic warning when do activities, exit actions, outgoing transitions, and/or nested state machines are removed by the compiler from final states.</h2>
```
Comment (TimLethbridge) [inline build/reference/9082ConflictingMethodModifers.txt]: This block is leftover from error 72 and is not correct in this file. Please change so it is correct.

**Review by TimLethbridge — CHANGES_REQUESTED** [review]
Please resolve the comment I left, and also deal with the merge conflict

**Review by TimLethbridge — APPROVED** [review]

**Comment by TimLethbridge** [comment]
Great job!

**Pull request merged**